### PR TITLE
Add new report generation and test#

### PR DESCRIPTION
**Pull request by awirb**
* Update report generation to be more flexible, now no particular keys are required but a type variable needs to be provided
* Add pytest for report generation

### CODE REVIEW

File: avaframe/com1DFA/com1DFA.py [hunk 1]
```diff
@@ -293,24 +294,25 @@ def runSamos(cfg, avaDir):
 
                 # Create dictionary
                 dictVar = {}
```
Comment (fso42) [inline avaframe/com1DFA/com1DFA.py:296]: Maybe rename dictVar dictNull etc to something like reportVar reportNull to make it obvious this is somehow related to the report generation

**Review by matthiasto — APPROVED** [review]

File: avaframe/log2Report/generateReport.py [hunk 2]
```diff
@@ -85,12 +109,12 @@ def writeReport(avaDir, reportDictList, plotDict='', reportOneFile=True):
             for reportD in reportDictList:
 
                 # extract additional info from log file
-                reportD['simParameters'].update(fU.extractParameterInfo(avaDir, reportD['simName']))
+                #reportD['simParameters'].update(fU.extractParameterInfo(avaDir, reportD['simName']))
```
Comment (fso42) [inline avaframe/log2Report/generateReport.py:112]: Just remove this line?

**Review by fso42 — CHANGES_REQUESTED** [review]
Please also update the report documentation. If you don't want to do it in this PR, just open an issue...

File: avaframe/com1DFA/com1DFA.py (hunk 1 above)
Comment (awirb) [reply avaframe/com1DFA/com1DFA.py:296]: dictionaries renamed to: reportVar, reportNull and reportST

File: avaframe/log2Report/generateReport.py (hunk 2 above)
Comment (awirb) [reply avaframe/log2Report/generateReport.py:112]: Done

**Comment by awirb** [comment]
I updated the documentation for log2Report

**Review by fso42 — APPROVED** [review]

**Pull request merged**